src/coreclr-debug/activate.ts

```diff
@@ -104,49 +107,65 @@ function showDotnetToolsWarning(message: string): void {
     }
 }
 
-interface AdapterExecutableCommand {
-    command: string;
-}
-
-// The default extension manifest calls this command as the adapterExecutableCommand
+// The activate method registers this factory to provide DebugAdapterDescriptors
 // If the debugger components have not finished downloading, the proxy displays an error message to the user
 // Else it will launch the debug adapter
-export async function getAdapterExecutionCommand(platformInfo: PlatformInformation, eventStream: EventStream, packageJSON: any, extensionPath: string): Promise<AdapterExecutableCommand> {
-    let util = new CoreClrDebugUtil(common.getExtensionPath());
-
-    // Check for .debugger folder. Handle if it does not exist.
-    if (!CoreClrDebugUtil.existsSync(util.debugAdapterDir())) {
-        // our install.complete file does not exist yet, meaning we have not completed the installation. Try to figure out what if anything the package manager is doing
-        // the order in which files are dealt with is this:
-        // 1. install.Begin is created
-        // 2. install.Lock is created
-        // 3. install.Begin is deleted
-        // 4. install.complete is created
-
-        // install.Lock does not exist, need to wait for packages to finish downloading.
-        let installLock = false;
-        let debuggerPackage = getRuntimeDependencyPackageWithId("Debugger", packageJSON, platformInfo, extensionPath);
-        if (debuggerPackage && debuggerPackage.installPath) {
-            installLock = await common.installFileExists(debuggerPackage.installPath, common.InstallFileType.Lock);
-        }
+export class DebugAdapterExecutableFactory implements vscode.DebugAdapterDescriptorFactory {
 
-        if (!installLock) {
-            eventStream.post(new DebuggerNotInstalledFailure());
-            throw new Error('The C# extension is still downloading packages. Please see progress in the output window below.');
-        }
-        // install.complete does not exist, check dotnetCLI to see if we can complete.
-        else if (!CoreClrDebugUtil.existsSync(util.installCompleteFilePath())) {
-            let success: boolean = await completeDebuggerInstall(platformInfo, eventStream);
+    constructor(private readonly platformInfo: PlatformInformation, private readonly eventStream: EventStream, private readonly packageJSON: any, private readonly extensionPath: string) {
```

Comment:
I'm not sure if this is a new Typescript feature, but are these arguments saved to the object?

Does `this.packageJSON`/`this.platformInfo` exist?

Reply:
It is a TypeScript shorthand. They are private fields.

Reply:
This is amazing! I've always assigned parameters to the object. Now to never have to write `self.parameter = parameter` in Typescript again.